common/flatpak-dir.c

```diff
@@ -1935,6 +1935,20 @@ flatpak_deploy_data_get_eol_rebase (GVariant *deploy_data)
   return eol;
 }
 
+const char **
+flatpak_deploy_data_get_previous_ids (GVariant *deploy_data)
```

Comment:
If you added a `gsize *length` argument to this function, you could avoid the `g_strv_length()` call and funny backwards `while` loop below.